layouts/partials/hooks/body-end.html

```diff
@@ -1,4 +1,25 @@
+{{/* We always include tracing.js in the build to avoid issue show up in the netlify build pipeline if an update of the otel modules breaks the current code */}}
 {{ if hugo.IsProduction -}}
-  {{ partial "script.html" (dict "src" "js/tracing.js") -}}
+  <script>
+    const collectorType = 'otlp';
+    const otlpTracesEndpointUrl = 'https://otelwebtelemetry.com/v1/traces';
+    const tracingEnabled = true;
+  </script>
+{{ else if eq (getenv "OTEL_TRACES_EXPORTER") "console" -}}
+<script>
+  const collectorType = 'console';
+  const tracingEnabled = true;
+</script>
+{{ else if getenv "OTEL_EXPORTER_OTLP_ENDPOINT" -}}
+<script>
+  const collectorType = 'otlp';
+  const otlpTracesEndpointUrl = '{{ getenv "OTEL_EXPORTER_OTLP_ENDPOINT" }}';
+  const tracingEnabled = true;
+</script>
+{{ else -}}
+<script>
+  const tracingEnabled = false;
+</script>
 {{ end -}}
```

Comment:
How about assuming production, and making it a little more dry? Something like this could replace lines 1 to 23, inclusive:

```html
<script>
  var tracingEnabled = true;
  var collectorType = 'otlp';
  var otlpTracesEndpointUrl = 'https://otelwebtelemetry.com/v1/traces';
{{ if eq (getenv "OTEL_TRACES_EXPORTER") "console" -}}
  collectorType = 'console';
{{ else if getenv "OTEL_EXPORTER_OTLP_ENDPOINT" -}}
  otlpTracesEndpointUrl = '{{ getenv "OTEL_EXPORTER_OTLP_ENDPOINT" }}';
{{ else if not hugo.IsProduction -}}
  tracingEnabled = false;
{{ end -}}
</script>
```

Reply:
yeah, that's a good idea, let me do that.

Reply:
applied